src/generator.rs

```diff
@@ -123,18 +131,27 @@ impl<'a> Generator<'a> {
         let mut body = String::new();
         html::push_html(&mut body, Parser::new(&ch.content));
 
-        let stylesheet_path = ch.path
-            .parent()
-            .expect("All chapters have a parent")
-            .components()
-            .map(|_| "..")
-            .chain(iter::once("stylesheet.css"))
-            .collect::<Vec<_>>()
-            .join("/");
-
-        let ctx = json!({ "title": ch.name, "body": body, "stylesheet": stylesheet_path });
-
-        self.hbs.render("index", &ctx)
+        match &ch.path {
```

Comment:
See my previous comment about using `ok_or_else()` and `?` to reduce indentation.

Reply:
fixed